src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -1435,31 +1458,27 @@ private static JsonWebToken ValidateSignature(string token, JsonWebToken jwtToke
             var exceptionStrings = new StringBuilder();
             var keysAttempted = new StringBuilder();
             var kidExists = !string.IsNullOrEmpty(jwtToken.Kid);
-            byte[] signatureBytes;
-
-            try
-            {
-                signatureBytes = Base64UrlEncoder.DecodeBytes(jwtToken.EncodedSignature);
-            }
-            catch (FormatException e)
-            {
-                throw new SecurityTokenInvalidSignatureException(TokenLogMessages.IDX10508, e);
-            }
 
             if (keys != null)
             {
                 foreach (var key in keys)
                 {
                     try
                     {
-                        if (ValidateSignature(encodedBytes, signatureBytes, key, jwtToken.Alg, jwtToken, validationParameters))
+#if NET45
+                        if (ValidateSignature(jwtToken.MessageBytes, jwtToken.SignatureBytes, key, jwtToken.Alg, jwtToken, validationParameters))
+#else
+                        if (ValidateSignature(jwtToken, key, validationParameters))
+#endif
                         {
-                            LogHelper.LogInformation(TokenLogMessages.IDX10242, token);
+                            LogHelper.LogInformation(TokenLogMessages.IDX10242, jwtToken.EncodedToken);
                             jwtToken.SigningKey = key;
                             return jwtToken;
                         }
                     }
+#pragma warning disable CA1031 // Do not catch general exception types
```

Comment:
>#pragma warning disable CA1031 [](http://example.com/codeflow?start=0&length=30)

nit: Looks like you're suppressing this in GlobalSuppressions and here, pattern seems to be just to do it in Global elsewhere for the library.

Reply:
I make the change to be consistent and put the suppressions in GolbalSuppressions.